### PR TITLE
Placement Review

### PR DESCRIPTION
**Pull request by mshuaibii**
Opening this PR to handle the review and cleanup of the placement code in this repo.

### CODE REVIEW

**Comment by abhshkdz** [comment]
Could you summarize the changes you've made please. Let's not have a separate `_refactor` subfolder, that's what git branches are for. Committing code to a subfolder makes this PR hard to review — there's a lot of new code and not easy to tell how much is duplicated vs. updated compared to the existing code.

File: _refactor/combined.py [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+        random_seed,
+        enumerate_all_configs,
+        added_z=2,
+        rotate_adsorbate=True,
+    ):
+        """
+        Adds adsorbate to surface, does the constraining, and aggregates all data necessary to write out.
+        Can either pick a random configuration or store all possible ones.
+
+        Args:
```
Comment (brookwander) [inline _refactor/combined.py:350]: partial docs?

File: _refactor/combined.py [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    """
+
+    def __init__(
+        self,
+        adsorbate,
+        surface,
+        random_sites,
+        random_seed,
+        enumerate_all_configs,
+        added_z=2,
```
Comment (brookwander) [inline _refactor/combined.py:343]: How is this done in catkit? This solution seems sloppy.

File: _refactor/combined.py [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+        }
+
+
+class CombinedRandomly:
+    """
+    This class handles all things with the adsorbate placed on a surface,
+    but the placement is random instead of heuristic.
+    Needs one adsorbate and one surface to create this class.
+
+    Attributes
```
Comment (brookwander) [inline _refactor/combined.py:306]: There appear to be unnecessary attributes

File: _refactor/combined.py [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+    ----------
+    adsorbate : Adsorbate
+        object representing the adsorbate
+    surface : Surface
+        object representing the surface
+    enumerate_all_configs : boolean
+        whether to enumerate all adslab placements instead of choosing one random
+    random_sites : int
```
Comment (brookwander) [inline _refactor/combined.py:314]: mentioned below as well but this variable is used twice

File: _refactor/combined.py [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+        Number of random placements
+    random_seed : int
+        A random seed
+    added_z: cushion space between surface atoms and adsorption site
+    rotate_adsorbate : boolean
+        Torsion rotate adsorbate, suggest default to True
+    adsorbed_surface_atoms : list
+        `Atoms` objects containing both the adsorbate and surface for all desired placements
+    adsorbed_surface_sampling_strs : list
```
Comment (brookwander) [inline _refactor/combined.py:323]: What is this used for? It seems arbitrary and maybe unnecessary

File: _refactor/combined.py [hunk 6]
```diff
@@ -0,0 +1,589 @@
...
+            triangle_positions = surface_atoms_pos[tri]
+            sites = self._random_sites_on_triangle(
+                triangle_positions, equal_distr, self.added_z
+            )
+            all_sites += sites
+        np.random.shuffle(all_sites)
+        return all_sites[:num_sites]
+
+    def _random_sites_on_triangle(self, surfsite_pos, k, added_z):
+        """
```
Comment (brookwander) [inline _refactor/combined.py:459]: partial doc

File: _refactor/combined.py [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+        Args:
+            adslab          An `ase.Atoms` object of the adsorbate+slab complex.
+
+        Returns:
+            A boolean indicating whether or not the adsorbate placement is
+            reasonable.
+        """
+        vnn = VoronoiNN(allow_pathological=True, tol=0.2, cutoff=10)
+        adsorbate_indices = [atom.index for atom in adslab if atom.tag >= 2]
+        adsorbate_bond_indices = [atom.index for atom in adslab if atom.tag == 3]
```
Comment (brookwander) [inline _refactor/combined.py:539]: where is this set to be 3?

File: _refactor/combined.py [hunk 8]
```diff
@@ -0,0 +1,589 @@
...
+                return False
+
+            slab_nn = [
+                nn for nn in nearneighbors if nn["site_index"] not in adsorbate_indices
+            ]
+            for nn in slab_nn:
+                ads_elem = structure[idx].species_string
+                nn_elem = structure[nn["site_index"]].species_string
+                cov_bond_thres = (
+                    0.8 * (COVALENT_RADIUS[ads_elem] + COVALENT_RADIUS[nn_elem]) / 100
```
Comment (brookwander) [inline _refactor/combined.py:558]: why was 0.8 chosen?

File: _refactor/combined.py [hunk 9]
```diff
@@ -0,0 +1,589 @@
...
+
+            adslab.cell = surface.cell
+            adslab.pbc = [True, True, False]
+            adslabs.append((site, adslab))
+        return adslabs
+
+    def is_config_reasonable(self, adslab):
+        """
+        Function that check whether the adsorbate placement is reasonable.
+        Two criteria are: 1. The adsorbate should be placed on the slab:
```
Comment (brookwander) [inline _refactor/combined.py:524]: I dont see criteria 1 implemented?

File: _refactor/combined.py (hunk 7 above)
Comment (janiceblue) [reply _refactor/combined.py:539]: `adsorbate_bond_indices` isn't actually used in this method so I think we're okay

File: _refactor/combined.py (hunk 9 above)
Comment (janiceblue) [reply _refactor/combined.py:524]: This seems to be a copy/paste of the heuristic `is_config_reasonable()` documentation. Is this criteria something we want to check for random placements, or was this intentionally taken out of the code?
